src/pages/search/Search.tsx

```diff
@@ -29,11 +31,65 @@ const Search = () => {
+        if (loading) {
+          return (
+            <Container tag={Fade} className={VIEWPORT_CENTER_LAYOUT}>
+              <Loading height="8rem" width="8rem" />
+            </Container>
+          );
+        }
+        if (error && error.graphQLErrors[0].extensions && error.graphQLErrors[0].extensions.code === 'BAD_USER_INPUT') {
+          return (
+            <Container tag={Fade} className="d-flex flex-column align-items-center justify-content-center bee-without-header-height-container p-0" fluid>
+              <div
+                className="bg-img-hero d-flex flex-column align-items-center justify-content-center gradient-overlay-half-primary-v1 bee-without-header-height-container w-100"
+                style={{ backgroundImage: `url('https://static.beenest.com/images/app/misc/painted-ladies2.jpg')` }}>
+                <Row className="px-4 mb-4" noGutters>
+                  <h5 className="font-size-md-down-5text-center text-white">Please enter in a city or region to see results.</h5>
+                </Row>
+                <Row className="d-flex justify-content-center px-0 mx-0 bg-white bee-top">
+                  <Col tag={Fade} className="w-100 p-5">
+                    <SearchBar />
+                  </Col>
+                </Row>
+              </div>
+            </Container>
+          );
+        } else if (error) {
+          return (
+            <Container tag={Fade} className={`${VIEWPORT_CENTER_LAYOUT} flex-column bee-without-header-height-container`}>
+              <h2 className="font-size-md-down-5 text-center">{getFriendlyErrorMessage(error)}</h2>
+              <h5 className="text-center">Please contact{' '}
+                <a className="text-primary" href="https://support.beenest.com/" target="_blank">support</a>{' '}
+                if this continues.
+              </h5>
+            </Container>
+          );
+        }
+
+        const { searchListings } = data;
+        if (searchListings.length === 0) {
+          return (
+            <Container tag={Fade} className="d-flex flex-column align-items-center justify-content-center bee-without-header-height-container p-0" fluid>
+            <div
+              className="bg-img-hero d-flex flex-column align-items-center justify-content-center gradient-overlay-half-primary-v1 bee-without-header-height-container w-100"
+              style={{ backgroundImage: `url('https://static.beenest.com/images/app/misc/painted-ladies2.jpg')` }}>
+              <Row className="px-4 mb-4" noGutters>
+                <h5 className="text-center text-white font-weight-normal">No listings in this area, please try a different city or region.</h5>
+              </Row>
+              <Row className="d-flex justify-content-center px-0 mx-0 bg-white bee-top">
+                <Col tag={Fade} className="w-100 p-5">
+                  <SearchBar />
+                </Col>
+              </Row>
+            </div>
+          </Container>
+          );
+        }
```

Comment:
Corollary to the above, could these be their own components? I'm picturing something approximately like:

    if (loading) {
      return <LoadingPage />;
    }
    if (error) {
      return isBadUserInputError(error) ? <SearchErrorPage /> : <ErrorPage error={error} />
    }
    if (searchListings.length < 1) {
      return <EmptySearchPage />;
    }
    return <SearchPage listing={searchListings} />;

I've been leaning on this as a more general pattern when using `Query`, as deep nesting can get hard to follow.

Reply:
great idea! will be much easier to navigate